src/codegen/generators/expression_generator.rs

```diff
@@ -638,7 +639,17 @@ impl<'ink, 'b> ExpressionCodeGenerator<'ink, 'b> {
                 //get paremeter at location
                 .get(location)
                 //find the parameter's type and name
-                .map(|it| Some((it.get_declaration_type(), it.get_type_name())))
+                .map(|it| {
+                    let name = it.get_type_name();
+                    if let Some(DataTypeInformation::Pointer {
+                        inner_type_name, ..
```

Comment:
Maybe we should only do this on auto deref pointers. It's an error to try and assign a value to the pointer otherwise